modepy/matrices.py

```diff
@@ -83,24 +92,66 @@ def vandermonde(functions, nodes):
     nnodes = nodes.shape[-1]
     nfunctions = len(functions)
 
+    if not functions:
+        return np.empty((nnodes, nfunctions), nodes.dtype)
+
+    f_iter = iter(functions)
+    f_first = next(f_iter)
+
+    f_first_values = f_first(nodes)
+
+    if isinstance(f_first_values, tuple):
+        warn("Calling vandermonde on tuple-returning functions is deprecated. "
+             "This will stop working in 2025. "
+             "Call multi_vandermonde instead.",
+             DeprecationWarning, stacklevel=2)
+        return multi_vandermonde(functions, nodes)
+
+    result = np.empty((nnodes, nfunctions), f_first_values.dtype)
+    result[:, 0] = f_first_values
+
+    for j, f in enumerate(functions):
+        result[:, j] = f(nodes)
+
+    return result
+
+
+def multi_vandermonde(
+            functions: Sequence[Callable[[np.ndarray], Sequence[np.ndarray]]],
+            nodes: np.ndarray
+        ) -> Sequence[np.ndarray]:
```

Comment:
```suggestion
        ) -> Tuple[np.ndarray, ...]:
```
? I'm never sure what rule to use here, but generally go for "generic inputs + specific outputs"?

Reply:
Done.